cmd/frontend/graphqlbackend/site_alerts.go

```diff
@@ -216,6 +216,10 @@ func updateAvailableAlert(args AlertFuncArgs) []*Alert {
 	if !args.ViewerFinalSettings.AlertsShowPatchUpdates && !isMinorUpdateAvailable(version.Version(), globalUpdateStatus.UpdateVersion) {
 		return nil
 	}
+	// for major/minor updates, ensure banner is hidden if they have opted out
+	if !args.ViewerFinalSettings.AlertsShowMajorMinorUpdates && isMinorUpdateAvailable(version.Version(), globalUpdateStatus.UpdateVersion) {
```

Comment:
```suggestion
	if !args.ViewerFinalSettings.AlertsShowMajorMinorUpdates {
```

I _think_ this should be a redundant check. nbd though.

Reply:
It was intentional. 

Even though the second conditional is always true, I found it easier to read, "if the alert is disabled and it is a major/minor update" as a whole sentence.

(just my own taste)